apps/ledger-live-mobile/src/screens/Settings/Debug/Features/FirmwareUpdate.tsx

```diff
@@ -1,48 +1,286 @@
-import React, { useState } from "react";
-import { Text, Flex, Button } from "@ledgerhq/native-ui";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { Subject } from "rxjs";
+import { Button, Flex, Slider, Switch, Text } from "@ledgerhq/native-ui";
+import { ScrollView } from "react-native-gesture-handler";
+import {
+  initialState,
+  UpdateFirmwareActionState,
+} from "@ledgerhq/live-common/deviceSDK/actions/updateFirmware";
+import { UpdateFirmwareTaskError } from "@ledgerhq/live-common/deviceSDK/tasks/updateFirmware";
+import { GetLatestFirmwareTaskError } from "@ledgerhq/live-common/deviceSDK/tasks/getLatestFirmware";
 import { Device } from "@ledgerhq/live-common/hw/actions/types";
-import { useUpdateFirmware } from "@ledgerhq/live-common/deviceSDK/hooks/useUpdateFirmware";
-import NavigationScrollView from "../../../../components/NavigationScrollView";
-import SelectDevice from "../../../../components/SelectDevice2";
+import { DeviceModelId } from "@ledgerhq/types-devices";
+import { FirmwareUpdateContext } from "@ledgerhq/types-live";
+
+import { FirmwareUpdate } from "../../../FirmwareUpdate";
+
+const mockFirmwareUpdateContext: FirmwareUpdateContext = {
+  final: {
+    application_versions: [],
+    bytes: 227672,
+    date_creation: "2022-06-27T08:53:35.663668Z",
+    date_last_modified: "2022-06-27T08:53:35.663668Z",
+    description: null,
+    device_versions: [17],
+    display_name: null,
+    firmware: "",
+    firmware_key: "",
+    hash: "",
+    id: 220,
+    mcu_versions: [91],
+    name: "2.0.2-il3",
+    notes: null,
+    osu_versions: [],
+    perso: "perso_11",
+    providers: [13],
+    se_firmware: 10,
+    version: "2.0.2",
+  },
+  osu: {
+    date_creation: "2022-06-27T08:53:37.277793Z",
+    date_last_modified: "2022-09-09T13:11:26.861355Z",
+    description: null,
+    device_versions: [17],
+    display_name: null,
+    firmware: "nanox/2.0.2-il3/fw_2.0.2-il0/upgrade_osu_2.0.2_il3",
+    firmware_key: "nanox/2.0.2-il3/fw_2.0.2-il0/upgrade_osu_2.0.2_il3_key",
+    hash: "",
+    id: 277,
+    name: "2.0.2-il3-osu",
+    next_se_firmware_final_version: 220,
+    notes: null,
+    perso: "perso_11",
+    previous_se_firmware_final_version: [200],
+    providers: [13],
+  },
+  shouldFlashMCU: true,
+};
+
+const mockDeviceInfo = {
+  bootloaderVersion: undefined,
+  hardwareVersion: undefined,
+  hasDevFirmware: true,
+  isBootloader: false,
+  isOSU: false,
+  isRecoveryMode: false,
+  languageId: undefined,
+  majMin: "2.0",
+  managerAllowed: true,
+  mcuBlVersion: undefined,
+  mcuTargetId: undefined,
+  mcuVersion: "2.90",
+  onboarded: true,
+  pinValidated: true,
+  providerName: null,
+  seTargetId: 857735172,
+  seVersion: "2.0.2-il0",
+  targetId: 857735172,
+  version: "2.0.2-il0",
+};
+
+const mockDevice: Device = {
+  deviceId: "it doesn't matter",
+  modelId: DeviceModelId.stax,
+  wired: false,
+};
 
 export default function DebugFirmwareUpdate() {
```

Comment:
It's really useful ! Nice !

We do agree that's it's only to test the UI components of the firmware update ? If so, update also the `<SettingRow desc="..">` of
`apps/ledger-live-mobile/src/screens/Settings/Debug/Features/index.tsx` because it's saying `"USE AT YOUR OWN RISK!"` ahaha

Reply:
Hahaha initially it was the actual update, but then I've changed to be UI-only and forgot to update that description 🙈 . Updated it now 👌